### PR TITLE
Get rid of RCP from eq()

### PR DESCRIPTION
**Pull request by certik**
`eq()` and `neq()` are not transferring the ownership, so they should not accept RCP arguments.

### CODE REVIEW

**Comment by certik** [comment]
Interesting, it compiled on my machine, but the expression test failed on Travis. Need to investigate this why it succeeded on my machine.

**Comment by certik** [comment]
Figured it out --- master got some new commits that broke this PR. I fixed it.

**Comment by certik** [comment]
I forgot about a test file. Now it should hopefully work.

**Comment by certik** [comment]
Looks like Travis actually succeeded.

@isuruf would you mind reviewing it?

File: symengine/tests/ntheory/test_ntheory.cpp [hunk 1]
```diff
@@ -580,7 +584,10 @@ TEST_CASE("test_nthroot_mod(): ntheory", "[ntheory]")
     nthroot_mod_list(roots, i9, i2, i27);
     REQUIRE(roots.size() == 6);
     v = {integer(3), integer(6), integer(12), integer(15), integer(21), integer(24)};
-    bool same = std::equal(v.begin(), v.end(), roots.begin(), SymEngine::eq);
+    bool same = std::equal(v.begin(), v.end(), roots.begin(),
+            [](const RCP<const Basic> &a, const RCP<const Basic> &b) {
```
Comment (isuruf) [inline symengine/tests/ntheory/test_ntheory.cpp]: We can use `RCPBasicKeyEq` here

**Comment by isuruf** [comment]
+1 to merge if tests pass

File: symengine/tests/ntheory/test_ntheory.cpp (hunk 1 above)
Comment (certik) [inline symengine/tests/ntheory/test_ntheory.cpp]: Good point, I'll do it here or in a new PR.

**Comment by certik** [comment]
Travis seems to be passing, Shippable fails for internal reasons.

File: symengine/tests/ntheory/test_ntheory.cpp (hunk 1 above)
Comment (certik) [inline symengine/tests/ntheory/test_ntheory.cpp]: I tried it, but it says:
```
/home/ondrej/repos/symengine/symengine/tests/ntheory/test_ntheory.cpp:589:37: error: expected primary-expression before ‘)’ token
             SymEngine::RCPBasicKeyEq);
                                     ^
```
I am not quite sure why. Any ideas?

Comment (isuruf) [inline symengine/tests/ntheory/test_ntheory.cpp]: You'll have to use `SymEngine::RCPBasicKeyEq()`

Comment (certik) [inline symengine/tests/ntheory/test_ntheory.cpp]: That works! I thought if you add (), that it calls the `operator()`, so I don't understand how it works, but it compiles and tests run.

My wifi doesn't allow me to push in changes, but I have it in a branch and push in later.

**Pull request merged**